shelley/chain-and-ledger/executable-spec/src/BlockChain.hs

```diff
@@ -107,13 +110,21 @@ bhHash = HashHeader . hash
 
 -- |Hash a given block body
 bbHash
-  :: Crypto crypto
+  :: forall crypto
+   . Crypto crypto
   => TxSeq crypto
   -> HashBBody crypto
-bbHash (TxSeq txns) = HashBBody $ Hash.hashPair (hash bodies) (hash wits)
+bbHash (TxSeq txns) = (HashBBody . unsafeCoerce) $ -- TODO is unsafeCoerce okay here?
```

Comment:
please let me know if there is a better/safer way to do this

Reply:
Yes! Friends don't let friends use `unsafePerformIO` or `unsafeCoerce`. There will undoubtedly be a better way, we just need to look. This is also non-optional, `unsafeCoerce` is wildly unsafe.

Reply:
it's gone now, @redxaxder helped me out :)